### PR TITLE
docs(auth): add usage docs

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

**Comment by lderrickable** [comment]
@damienwebdev and I were talking about how providing custom drivers and transformers would work, and it turns out that the driver service almost never needs to be overridden. Usually the only thing an app dev will want that is custom is specific data from a backend platform, which means they need only custom models and custom transformers. So as long as there exists generic transformer interfaces and a generic interface and instance of a driver, the app dev should only need to override the transformer. I just mention this here, because it might make more sense for the readme to say that they usually only need to override the transformers and extend the daffodil models.

**Comment by griest024** [comment]
@lderrickable Should I mention the service customization as an optional step or omit it entirely?

**Comment by lderrickable** [comment]
TLDR: no just omit it.

I think we omit for now, because if they're overriding the driver service, they should have a really good idea what they're doing. If they override the service (presumably to make a completely different service call), then I'm not sure they'll even be able to use the redux state we provide as it exists now. Damien has been talking about possibly getting rid of the actions and effects layer in daffodil and allow the user to provide that for better action hygiene. And maybe there'll be a more refined process for providing a custom driver, but for now it doesn't really make sense to override it.

File: libs/auth/README.md [hunk 1]
```diff
@@ -12,3 +12,35 @@ npm install @daffodil/auth \
...
+### Login
+If the `DaffLoginResponse` doesn't have all of the necessary fields you require, you'll need to provide two things.
+1. A custom service that implements the `DaffAuthTransformerService`. This should be injected with the `DaffAuthTransformer` injection token to override the default transformer.
+2. Define a model that extends the `DaffLoginResponse` and/or the `DaffLoginRequest`.
+
+These steps should allow you to make supported requests to login in whatever way you'd like, while still being compatible with the daffodil redux state, because all redux components are generically defined.
+
+### Register
+If the `DaffRegisterResponse` doesn't have all of the necessary fields you require, you'll need to provide two things.
+1. A custom service that implements the `DaffAuthTransformerService`. This should be injected with the `DaffAuthTransformer` injection token to override the default transformer.
```
Comment (lderrickable) [inline libs/auth/README.md]: This seems wrong. Both login and register don't share the same transformer token and interface, right?

Comment (griest024) [reply libs/auth/README.md]: `register` calls `login` and returns whatever `login` returns, so in this case, they do "share" it. However since `register` doesn't explicitly invoke the transformer, its probably not correct to talk about it in the `register` section here.

It might be useful to move this after the driver interface PR.

**Comment by damienwebdev** [comment]
I don't think that this is sufficient docs to explain the use case.
